### PR TITLE
tests, Move VMI lifecycle networking tests to network folder

### PR DESCRIPTION
**Pull request by oshoval**
Move the two networking specific tests of `VMIlifecycle`
to network tests folder.
This way those tests will be considered as `sig-network`,
and maintained by `sig-network` group.
`sig-network` maintains the production code related to bridge binding,
and also those tests, which cover bridge binding usage scenarios.

The tests were introduced by PR [1],
and were added to `vmi_lifecycle_test.go`
but they are actually network feature specific (bridge binding), 
and should have been part of the networking folder.
Selection of `vmi_lifecycle_test.go` was 
suitable when the PR was opened, but now we have a network folder.

Explanation about the tests:

With bridge binding mode, we move the pod IP address to VM
and the pod interface is left IP-less.

In case the CNI implements a CHECK RPC call,
`kubelet`, after its restart, it will perform an IP check and since there isn't an IP
it will result in pod removal.

The PR [1] introduces a fix for that, by creating a dummy interface,
named after the original NIC, and which isn't connected to any network.
This dummy interface holds the IP, so the check won't cause pod removal.

The interface will ignore ARP requests, in order to not be detected by Duplicate Address Detection (DAD),
which otherwise will cause the VM to lose its IP, due to duplication.

The PR [1] also introduced the two tests that are moved by this PR.
The 1st test restarts the `kubelet` and checks this flow.
The 2nd test checks that Duplicate Address Detection (DAD),
won't detect this duplicate IP.

`EnsureKVMPresent` was removed as the virt-handler is not restarted
by those tests.

[1] https://github.com/kubevirt/kubevirt/pull/4552
```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
/cc @EdDev @phoracek 

still testing it, but ready for review (tested locally)

**Comment by EdDev** [comment]
I suggest you focus in this PR only on moving the tests, and not forcing the refactoring here.
You can copy the needed code and in a following PR suggest solving it using refactoring. What you ended up here is touching many file tests instead of the one that the code is to be extracted from and the one you add it to.

In addition, moving the code to that all-in-one ambiguous file will inflate it even more with no good reason. Try to think of proper other file names or just call it the same as the original.

It will also be nice to reason better on why these tests belong to the network sig.

**Comment by oshoval** [comment]
> I suggest you focus in this PR only on moving the tests, and not forcing the refactoring here.
> You can copy the needed code and in a following PR suggest solving it using refactoring. What you ended up here is touching many file tests instead of the one that the code is to be extracted from and the one you add it to.

This is why it was split to commits, but NP i will start as you said.

> It will also be nice to reason better on why these tests belong to the network sig.

Discussing with Petr, those two tests were there because they are indeed network related, and catched in the past real bugs,
having them in sig-network, as they were part of CNAO, is for sanity checking of the VMI lifecycle where CNAO is deployed.

**Comment by EdDev** [comment]
> Discussing with Petr, those two tests were there because they are indeed network (bridge) related, and caught in the past real bugs, having them in sig-network, as they were part of CNAO, is for sanity checking of the VMI lifecycle where CNAO is deployed.

They are part of both jobs, right? We need a technical explanation of why they belong to the sig-network, unrelated to which jobs ran them before. What you reasoned for above are circumstances of the current state, we need and explanation that a future reader of the history will understand why it fits under the sig-network and not in a general vmi-lifecycle (sig-virt?) one.

**Comment by oshoval** [comment]
Dropped the refactors.

**Comment by oshoval** [comment]
pull-kubevirt-unit-test not related to this PR

**Comment by oshoval** [comment]
Updated the PR description, when its ok i will update the commit message as well

**Review by EdDev — CHANGES_REQUESTED** [review]
Thank you for explaining why these tests belong to the network sig.

> In addition, moving the code to that all-in-one ambiguous file will inflate it even more with no good reason. Try to think of proper other file names or just call it the same as the original.

You may have missed this comment form before.

**Comment by oshoval** [comment]
> > In addition, moving the code to that all-in-one ambiguous file will inflate it even more with no good reason. Try to think of proper other file names or just call it the same as the original.
> 
> You may have missed this comment form before.

It was about utils.go isn't it ?
if so i dropped the refactor and this will be relevant on a following PR.
or did you mean the vmi_networking.go file?

**Comment by oshoval** [comment]
Addressed comments

File: tests/network/BUILD.bazel [hunk 1]
```diff
@@ -12,6 +12,7 @@ go_library(
         "services.go",
         "vmi_multus.go",
         "vmi_networking.go",
+        "vmi_networking_lifecycle.go",
```
Comment (EdDev) [inline tests/network/BUILD.bazel:15]: We are under the `network` folder, I do not think we need `networking` in the name.

File: tests/network/vmi_networking_lifecycle.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
Comment (EdDev) [inline tests/network/vmi_networking_lifecycle.go:28]: These seem to be two distinct groups. Please separate `goexpect` and the `onsi` ones from the rest.

**Review by EdDev — CHANGES_REQUESTED** [review]
Thanks.
There are two small comments inline.

**Comment by oshoval** [comment]
Addressed comments

File: tests/network/vmi_lifecycle.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	expect "github.com/google/goexpect"
```
Comment (EdDev) [inline tests/network/vmi_lifecycle.go:26]: This has not been separated.

Comment (oshoval) [reply tests/network/vmi_lifecycle.go:26]: Done, just please note that if the file changes the `make` / `make build-functests` might remove spaces

**Comment by oshoval** [comment]
Addressed comments

**Review by EdDev — APPROVED** [review]
Thanks

**Comment by oshoval** [comment]
Thanks Edy

/cc @AlonaKaplan 
can you take a look please?

**Comment by oshoval** [comment]
Thanks

/retest

**Pull request merged**